attestation-agent/attester/src/cca/mod.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2023 Arm Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::Attester;
+use anyhow::*;
+use base64::Engine;
+use nix::fcntl::{open, OFlag};
+use nix::sys::stat::Mode;
+use nix::unistd::close;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+// NOTE: The path might be different when the CCA feature is public available, will come back to update the actual path if needed.
+pub fn detect_platform() -> bool {
+    Path::new("/dev/cca_attestation").exists()
+}
+
+#[derive(Debug, Default)]
+pub struct CCAAttester {}
+
+#[derive(Serialize, Deserialize)]
+struct CcaEvidence {
+    // Cca token
+    token: Vec<u8>,
+}
+
+#[allow(non_camel_case_types)]
+#[repr(C)]
+pub struct cca_ioctl_request {
+    challenge: [u8; 64],
+    token: [u8; 4096],
+    token_length: u64,
+}
+
+nix::ioctl_readwrite!(cca_attestation_request, b'A', 1, cca_ioctl_request);
+
+#[async_trait::async_trait]
+impl Attester for CCAAttester {
+    async fn get_evidence(&self, _: Vec<u8>) -> Result<String> {
+        let challenge = [0u8; 64];
+        let token = attestation(&challenge, 0).unwrap();
+        let evidence = CcaEvidence { token };
+        serde_json::to_string(&evidence)
+            .map_err(|e| anyhow!("Serialize CCA evidence failed: {:?}", e))
```

Comment:
Another choice is
```suggestion
        let ev = serde_json::to_string(&evidence).context("Serialize CCA evidence failed")?;
        Ok(ev)
```

Which is easy to make breakpoints when doing single step debug : )

Reply:
Done